src/ucp/rndv/rndv.h

```diff
@@ -95,13 +91,10 @@ ucs_status_t ucp_rndv_rts_handle_status_from_pending(ucp_request_t *sreq,
 static UCS_F_ALWAYS_INLINE int
 ucp_rndv_rts_is_tag(const ucp_rndv_rts_hdr_t *rts_hdr)
 {
-    return rts_hdr->flags & UCP_RNDV_RTS_FLAG_TAG ||
-           /* Wire compatibility with old versions of RNDV protocol which uses
-            * status instead of flags for TAG operations */
-           ucs_unlikely((rts_hdr->flags ==
-                         UCP_RNDV_RTS_FLAG_TAG_COMPAT_VALUE_OK) ||
-                        (rts_hdr->flags ==
-                         UCP_RNDV_RTS_FLAG_TAG_COMPAT_VALUE_CANCELED));
+    ucs_assert((rts_hdr->opcode == UCP_RNDV_RTS_TAG_OK) ||
+               (rts_hdr->opcode == UCP_RNDV_RTS_TAG_CANCELED) ||
+               (rts_hdr->opcode == UCP_RNDV_RTS_AM));
```

Comment:
==TAG_OK || ==TAG_CANCELED

Reply:
so we can inverse the condition in the caller functions, but is_tag should check for tag explicitly

Reply:
done